dnf/plugin.py

```diff
@@ -193,13 +194,18 @@ def unload_removed_plugins(self, transaction):
 
         # check whether removed plugin file is added at the same time (upgrade of a plugin)
         for pkg in transaction.install_set:
-            erased_plugin_files.difference_update(pkg.files)
+            erased_plugin_files.difference_update(_get_files_from_newly_installed_pkg(pkg))
 
         # unload plugins that were removed in transaction
         for plugin_file in erased_plugin_files:
             self.plugins.remove(plugins[plugin_file])
 
 
+def _get_files_from_newly_installed_pkg(pkg):
+    hdr = dnf.util.first(pkg.base._ts.dbMatch('name', pkg.name))
```

Comment:
Wouldn't it be better to do `pkg.get_header()`?

Reply:
There is a problem in the `get_header()` method:

```python
def get_header(self):
        """
        Returns the rpm header of the package if it is installed. If not
        installed, returns None. The header is not cached, it is retrieved from
        rpmdb on every call. In case of a failure (e.g. when the rpmdb changes
        between loading the data and calling this method), raises an instance
        of PackageNotFoundError.
        """
        if not self._from_system:
            return None
```

In our case, the package is not from system repo, so it returns `None`.

Reply:
Maybe we can use `pkg._header[rpm.RPMTAG_FILENAMES]`...